distributed/scheduler.py

```diff
@@ -1590,6 +1596,33 @@ def get_nbytes(self, stream=None, keys=None, summary=True):
 
             return result
 
+    def publish_dataset(self, stream=None, keys=None, data=None, name=None,
+                        client=None):
+        self.client_wants_keys(keys, 'published-%s' % name)
+        self.published_data[name] = {'data': data, 'keys': keys,
+                                     'clients': {client}}
```

Comment:
Yes, that would be good.  I recommend the response `'OK'` or `{'status': 'OK'}` as that is used in a few other places.

You could also raise an error.  I *think* that all of the pieces are in place to propagate that error from the scheduler coroutine to the caller.  I could be wrong though.